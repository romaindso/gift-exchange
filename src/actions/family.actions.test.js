
import {
  addFamilyMember,
  removeFamilyMember,
  shuffleFamilyMembers,
  resetState
} from "./family.actions";

describe('Family actions', () => {

  test('ADD_FAMILY_MEMBER', () => {
    const familyMember = {
      name: "Romain",
      spouse: "Amélie"
    };

    expect(addFamilyMember(familyMember)).toMatchSnapshot();
  });

  test('REMOVE_FAMILY_MEMBER', () => {
    const familyMember = {
      name: "Romain",
      spouse: "Amélie"
    };

    expect(removeFamilyMember(familyMember)).toMatchSnapshot();
  });

  test('SHUFFLE_FAMILY_MEMBERS', () => {
    expect(shuffleFamilyMembers()).toMatchSnapshot();
  });

  test('RESET_STATE', () => {
    expect(resetState()).toMatchSnapshot();
  });

});