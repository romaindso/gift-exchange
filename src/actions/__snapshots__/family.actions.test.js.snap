// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Family actions ADD_FAMILY_MEMBER 1`] = `
Object {
  "familyMember": Object {
    "name": "Romain",
    "spouse": "Amélie",
  },
  "type": "ADD_FAMILY_MEMBER",
}
`;

exports[`Family actions REMOVE_FAMILY_MEMBER 1`] = `
Object {
  "familyMember": Object {
    "name": "Romain",
    "spouse": "Amélie",
  },
  "type": "REMOVE_FAMILY_MEMBER",
}
`;

exports[`Family actions RESET_STATE 1`] = `
Object {
  "type": "RESET_STATE",
}
`;

exports[`Family actions SHUFFLE_FAMILY_MEMBERS 1`] = `
Object {
  "type": "SHUFFLE_FAMILY_MEMBERS",
}
`;
